packages/react-sdk/src/core/pantheon-api.ts

```diff
@@ -90,10 +90,10 @@ export function PantheonAPI(options?: PantheonAPIOptions) {
             )
           : null;
 
-      if (article == null) {
-        article = {
-          id: parsedArticleId,
-        };
+      if (!article?.id) {
+        throw new Error(
+          `Article with ID or slug "${parsedArticleId}" not found`,
+        );
       }
```

Comment:
We should redirect to a 404 page in this case. If we just throw an error I think they will see an internal server error. Maybe PantheonAPI should have a handleError kind of function so that it can do whatever it wants with a 404 or 500 errors.

Reply:
Keeping the prior behaviour here, if it's unavailable we'll redirect to their article site and let them show the article not found state.